CDC_Backend/APIs/studentViews.py

```diff
@@ -277,19 +322,86 @@ def getContributorStats(request, id, email, user_type):
         
 #view for sudentAcceptOffer
 @api_view(['POST'])
+@precheck(required_data=[OPENING_ID,"offer_accepted"])  
 @isAuthorized(allowed_users=[STUDENT])
 def studentAcceptOffer(request, id, email, user_type):
     try:
-        company_id = request.data['id']
-        student_id=request.data['profileInfo']['id']
-        offer_status = request.data['offerStatus']
-        placement_application=PlacementApplication.objects.get(placement=company_id,student=student_id)
-        placement_application.offer_accepted=offer_status
-        placement_application.save()
-        return Response({'action': "Accept Offer", 'message': "Updated Offer Status"},
+        company_id = request.data[OPENING_ID]
+        #student_id=request.data['profileInfo']['id']             //check this once
```

Comment:
As The endpoint is Specific to a Student id can be extracted directly without getting it from request